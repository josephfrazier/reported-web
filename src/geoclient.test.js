/**
 * @jest-environment node
 */
/* eslint-env jest */

import { validateLocation, processValidation } from './geoclient.js';

describe('validateLocation', () => {
  test('returns the right object', async () => {
    const result = await validateLocation({
      lat: 40.7128,
      long: -74.006,
    });

    expect(result).toMatchSnapshot();
  });
});

describe('processValidation', () => {
  test('returns the right object', async () => {
    const result = await processValidation({
      lat: 40.7128,
      long: -74.006,
    });

    expect(result).toMatchSnapshot();
  });

  test('returns the right object for string lat/long', async () => {
    const result = await processValidation({
      lat: '40.772204329580426',
      long: '-73.92590668144963',
    });

    expect(result).toMatchSnapshot();
  });
});
