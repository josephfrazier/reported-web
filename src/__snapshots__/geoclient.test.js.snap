// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processValidation returns the right object 1`] = `
Object {
  "cb_data": Object {
    "address": "49-51 Chambers Street, Room 715
New York, NY 10007",
    "borough": "Manhattan",
    "email": "man01@cb.nyc.gov",
    "fax": "212-442-5055",
    "id": 101,
    "neighborhoods": "Tribeca, Seaport/Civic Center, Financial District, Battery Park City",
    "number": 1,
    "phone": "212-442-5050",
    "twitter": "CommunityBoard1",
    "website": "http://www.nyc.gov/html/mancb1/html/home/home.shtml",
  },
  "geoclient_response": Object {
    "address": Object {
      "assemblyDistrict": "66",
      "bbl": "1001237501",
      "bblBoroughCode": "1",
      "bblTaxBlock": "00123",
      "bblTaxLot": "7501",
      "boardOfElectionsPreferredLgc": "1",
      "boePreferredStreetName": "BROADWAY",
      "boePreferredstreetCode": "11361001",
      "boroughCode1In": "1",
      "buildingIdentificationNumber": "1087167",
      "businessImprovementDistrict": "113140",
      "censusBlock2000": "1008",
      "censusBlock2010": "1008",
      "censusTract1990": "  21  ",
      "censusTract2000": "  21  ",
      "censusTract2010": "  21  ",
      "cityCouncilDistrict": "01",
      "civilCourtDistrict": "01",
      "coincidentSegmentCount": "1",
      "communityDistrict": "101",
      "communityDistrictBoroughCode": "1",
      "communityDistrictNumber": "01",
      "communitySchoolDistrict": "02",
      "condominiumBillingBbl": "1001237501",
      "condominiumFlag": "C",
      "congressionalDistrict": "10",
      "continuousParityIndicator1a": "L",
      "continuousParityIndicator1e": "L",
      "cooperativeIdNumber": "0000",
      "crossStreetNamesFlagIn": "E",
      "dcpPreferredLgc": "01",
      "dofCondominiumIdentificationNumber": "1514",
      "dotStreetLightContractorArea": "1",
      "dynamicBlock": "109",
      "electionDistrict": "004",
      "fireBattalion": "01",
      "fireCompanyNumber": "010",
      "fireCompanyType": "L",
      "fireDivision": "01",
      "firstBoroughName": "MANHATTAN",
      "firstStreetCode": "11361001010",
      "firstStreetNameNormalized": "BROADWAY",
      "fromLionNodeId": "0015354",
      "fromPreferredLgcsFirstSetOf5": "0101",
      "genericId": "0000386",
      "geosupportFunctionCode": "1B",
      "geosupportReturnCode": "00",
      "geosupportReturnCode2": "00",
      "gi5DigitStreetCode1": "13610",
      "gi5DigitStreetCode2": "27880",
      "gi5DigitStreetCode3": "12520",
      "giBoroughCode1": "1",
      "giBoroughCode2": "1",
      "giBoroughCode3": "1",
      "giBuildingIdentificationNumber1": "1087167",
      "giBuildingIdentificationNumber2": "1087167",
      "giBuildingIdentificationNumber3": "1087167",
      "giDcpPreferredLgc1": "01",
      "giDcpPreferredLgc2": "01",
      "giDcpPreferredLgc3": "01",
      "giHighHouseNumber1": "237",
      "giHighHouseNumber2": "20",
      "giHighHouseNumber3": "19",
      "giLowHouseNumber1": "229",
      "giLowHouseNumber2": "2",
      "giLowHouseNumber3": "1",
      "giSideOfStreetIndicator1": "L",
      "giSideOfStreetIndicator2": "L",
      "giSideOfStreetIndicator3": "R",
      "giStreetCode1": "11361001",
      "giStreetCode2": "12788001",
      "giStreetCode3": "11252001",
      "giStreetName1": "BROADWAY",
      "giStreetName2": "PARK PLACE",
      "giStreetName3": "BARCLAY STREET",
      "healthArea": "7700",
      "healthCenterDistrict": "15",
      "highBblOfThisBuildingsCondominiumUnits": "1001231013",
      "highCrossStreetB5SC1": "127880",
      "highCrossStreetCode1": "12788001",
      "highCrossStreetName1": "PARK PLACE",
      "highHouseNumberOfBlockfaceSortFormat": "000238000AA",
      "houseNumber": "230",
      "houseNumberIn": "230",
      "houseNumberSortFormat": "000230000AA",
      "hurricaneEvacuationZone": "5",
      "instructionalRegion": "MS",
      "interimAssistanceEligibilityIndicator": "I",
      "internalLabelXCoordinate": "0981941",
      "internalLabelYCoordinate": "0198844",
      "latitude": 40.7121526074704,
      "latitudeInternalLabel": 40.7124572517608,
      "legacySegmentId": "0023364",
      "lionBoroughCode": "1",
      "lionBoroughCodeForVanityAddress": "1",
      "lionFaceCode": "0755",
      "lionFaceCodeForVanityAddress": "0755",
      "lionKey": "1075501170",
      "lionKeyForVanityAddress": "1075501170",
      "lionSequenceNumber": "01170",
      "lionSequenceNumberForVanityAddress": "01170",
      "listOf4Lgcs": "0107",
      "longitude": -74.00798614053389,
      "longitudeInternalLabel": -74.0083288540082,
      "lowBblOfThisBuildingsCondominiumUnits": "1001231001",
      "lowCrossStreetB5SC1": "112520",
      "lowCrossStreetB5SC2": "127890",
      "lowCrossStreetCode1": "11252001",
      "lowCrossStreetCode2": "12789001",
      "lowCrossStreetName1": "BARCLAY STREET",
      "lowCrossStreetName2": "PARK ROW",
      "lowHouseNumberOfBlockfaceSortFormat": "000228000AA",
      "lowHouseNumberOfDefiningAddressRange": "000230000AA",
      "nta": "MN24",
      "ntaName": "SoHo-TriBeCa-Civic Center-Little Italy",
      "numberOfCrossStreetB5SCsHighAddressEnd": "1",
      "numberOfCrossStreetB5SCsLowAddressEnd": "2",
      "numberOfCrossStreetsHighAddressEnd": "1",
      "numberOfCrossStreetsLowAddressEnd": "2",
      "numberOfEntriesInListOfGeographicIdentifiers": "0003",
      "numberOfExistingStructuresOnLot": "0001",
      "numberOfStreetFrontagesOfLot": "03",
      "physicalId": "0000499",
      "policePatrolBoroughCommand": "1",
      "policePrecinct": "001",
      "returnCode1a": "00",
      "returnCode1e": "00",
      "roadwayType": "1",
      "rpadBuildingClassificationCode": "RB",
      "rpadSelfCheckCodeForBbl": "0",
      "sanbornBoroughCode": "1",
      "sanbornPageNumber": "019",
      "sanbornVolumeNumber": "1S",
      "sanitationCollectionSchedulingSectionAndSubsection": "1A",
      "sanitationDistrict": "101",
      "sanitationRecyclingCollectionSchedule": "EF",
      "sanitationRegularCollectionSchedule": "MWF",
      "sanitationSnowPriorityCode": "C",
      "segmentAzimuth": "058",
      "segmentIdentifier": "0023364",
      "segmentLengthInFeet": "00230",
      "segmentOrientation": "N",
      "segmentTypeCode": "U",
      "selfCheckCodeOfBillingBbl": "0",
      "sideOfStreetIndicator": "L",
      "sideOfStreetOfVanityAddress": "L",
      "splitLowHouseNumber": "000228000AA",
      "stateSenatorialDistrict": "26",
      "streetName1In": "BROADWAY",
      "streetStatus": "2",
      "streetWidth": "52",
      "taxMapNumberSectionAndVolume": "10104",
      "toLionNodeId": "0015357",
      "toPreferredLgcsFirstSetOf5": "01",
      "trafficDirection": "A",
      "underlyingStreetCode": "11361001",
      "uspsPreferredCityName": "NEW YORK",
      "workAreaFormatIndicatorIn": "C",
      "xCoordinate": "0982036",
      "xCoordinateHighAddressEnd": "0982125",
      "xCoordinateLowAddressEnd": "0982004",
      "xCoordinateOfCenterofCurvature": "0000000",
      "yCoordinate": "0198733",
      "yCoordinateHighAddressEnd": "0198871",
      "yCoordinateLowAddressEnd": "0198675",
      "yCoordinateOfCenterofCurvature": "0000000",
      "zipCode": "10279",
    },
  },
  "google_response": Object {
    "plus_code": Object {
      "compound_code": "PX7V+4H New York, NY, USA",
      "global_code": "87G7PX7V+4H",
    },
    "results": Array [
      Object {
        "address_components": Array [
          Object {
            "long_name": "230",
            "short_name": "230",
            "types": Array [
              "street_number",
            ],
          },
          Object {
            "long_name": "Broadway",
            "short_name": "Broadway",
            "types": Array [
              "route",
            ],
          },
          Object {
            "long_name": "Manhattan",
            "short_name": "Manhattan",
            "types": Array [
              "political",
              "sublocality",
              "sublocality_level_1",
            ],
          },
          Object {
            "long_name": "New York",
            "short_name": "New York",
            "types": Array [
              "locality",
              "political",
            ],
          },
          Object {
            "long_name": "New York County",
            "short_name": "New York County",
            "types": Array [
              "administrative_area_level_2",
              "political",
            ],
          },
          Object {
            "long_name": "New York",
            "short_name": "NY",
            "types": Array [
              "administrative_area_level_1",
              "political",
            ],
          },
          Object {
            "long_name": "United States",
            "short_name": "US",
            "types": Array [
              "country",
              "political",
            ],
          },
          Object {
            "long_name": "10007",
            "short_name": "10007",
            "types": Array [
              "postal_code",
            ],
          },
        ],
        "formatted_address": "230 Broadway, New York, NY 10007, USA",
        "geometry": Object {
          "location": Object {
            "lat": 40.7127421,
            "lng": -74.00596890000001,
          },
          "location_type": "ROOFTOP",
          "viewport": Object {
            "northeast": Object {
              "lat": 40.7140910802915,
              "lng": -74.00461991970852,
            },
            "southwest": Object {
              "lat": 40.7113931197085,
              "lng": -74.00731788029152,
            },
          },
        },
        "place_id": "ChIJVzXOjBhawokRSp5VqCM4F9Q",
        "plus_code": Object {
          "compound_code": "PX7V+3J New York, United States",
          "global_code": "87G7PX7V+3J",
        },
        "types": Array [
          "street_address",
        ],
      },
    ],
    "status": "OK",
  },
  "lat": 40.7128,
  "long": -74.006,
  "valid": true,
}
`;

exports[`processValidation returns the right object for string lat/long 1`] = `
Object {
  "cb_data": Object {
    "address": "45-02 Ditmars Boulevard
LL Suite 125
Astoria, NY 11105",
    "borough": "Queens",
    "calendar": Object {
      "web": "http://www1.nyc.gov/site/queenscb1/calendar/calendar.page",
    },
    "email": "qn01@cb.nyc.gov",
    "fax": "718-626-1072",
    "id": 401,
    "neighborhoods": "Astoria, Old Astoria, Long Island City, Queensbridge, Ditmars, Ravenswood, Steinway, Garden Bay, and Woodside",
    "number": 1,
    "phone": "718-626-1021",
    "twitter": "cb1queens",
    "website": "http://www.nyc.gov/html/qnscb1/html/home/home.shtml",
  },
  "geoclient_response": Object {
    "address": Object {
      "assemblyDistrict": "36",
      "bbl": "4005430015",
      "bblBoroughCode": "4",
      "bblTaxBlock": "00543",
      "bblTaxLot": "0015",
      "boardOfElectionsPreferredLgc": "1",
      "boePreferredStreetName": "21 STREET",
      "boePreferredstreetCode": "40674001",
      "boroughCode1In": "4",
      "buildingIdentificationNumber": "4458141",
      "censusBlock2000": "1000",
      "censusBlock2010": "1000",
      "censusTract1990": "  71  ",
      "censusTract2000": "  71  ",
      "censusTract2010": "  71  ",
      "cityCouncilDistrict": "22",
      "civilCourtDistrict": "01",
      "coincidentSegmentCount": "1",
      "communityDistrict": "401",
      "communityDistrictBoroughCode": "4",
      "communityDistrictNumber": "01",
      "communitySchoolDistrict": "30",
      "condominiumBillingBbl": "0000000000",
      "congressionalDistrict": "12",
      "cornerCode": "SE",
      "crossStreetNamesFlagIn": "E",
      "dcpCommercialStudyArea": "43003",
      "dcpPreferredLgc": "01",
      "dotStreetLightContractorArea": "4",
      "dynamicBlock": "101",
      "electionDistrict": "045",
      "fireBattalion": "49",
      "fireCompanyNumber": "262",
      "fireCompanyType": "E",
      "fireDivision": "14",
      "firstBoroughName": "QUEENS",
      "firstStreetCode": "40674001010",
      "firstStreetNameNormalized": "21 STREET",
      "fromLionNodeId": "0042176",
      "fromPreferredLgcsFirstSetOf5": "010101",
      "genericId": "0009841",
      "geosupportFunctionCode": "1B",
      "geosupportReturnCode": "00",
      "geosupportReturnCode2": "01",
      "gi5DigitStreetCode1": "06740",
      "gi5DigitStreetCode2": "57390",
      "giBoroughCode1": "4",
      "giBoroughCode2": "4",
      "giBuildingIdentificationNumber1": "4458141",
      "giBuildingIdentificationNumber2": "4458141",
      "giDcpPreferredLgc1": "01",
      "giDcpPreferredLgc2": "01",
      "giHighHouseNumber1": "27-11",
      "giHighHouseNumber2": "21-04",
      "giLowHouseNumber1": "27-01",
      "giLowHouseNumber2": "21-02",
      "giSideOfStreetIndicator1": "L",
      "giSideOfStreetIndicator2": "R",
      "giStreetCode1": "40674001",
      "giStreetCode2": "45739001",
      "giStreetName1": "21 STREET",
      "giStreetName2": "NEWTOWN AVENUE",
      "healthArea": "0300",
      "healthCenterDistrict": "41",
      "highBblOfThisBuildingsCondominiumUnits": "4005430015",
      "highCrossStreetB5SC1": "407450",
      "highCrossStreetCode1": "40745001",
      "highCrossStreetName1": "27 ROAD",
      "highHouseNumberOfBlockfaceSortFormat": "100027019AA",
      "houseNumber": "27-01",
      "houseNumberIn": "2701",
      "houseNumberSortFormat": "100027001AA",
      "hurricaneEvacuationZone": "6",
      "instructionalRegion": "QN",
      "interimAssistanceEligibilityIndicator": "I",
      "internalLabelXCoordinate": "1004768",
      "internalLabelYCoordinate": "0220510",
      "latitude": 40.77216228771581,
      "latitudeInternalLabel": 40.77190150745826,
      "legacySegmentId": "0068622",
      "lionBoroughCode": "4",
      "lionBoroughCodeForVanityAddress": "4",
      "lionFaceCode": "5256",
      "lionFaceCodeForVanityAddress": "5256",
      "lionKey": "4525600225",
      "lionKeyForVanityAddress": "4525600225",
      "lionSequenceNumber": "00225",
      "lionSequenceNumberForVanityAddress": "00225",
      "listOf4Lgcs": "01",
      "longitude": -73.92596956044154,
      "longitudeInternalLabel": -73.92592291582544,
      "lowBblOfThisBuildingsCondominiumUnits": "4005430015",
      "lowCrossStreetB5SC1": "407440",
      "lowCrossStreetB5SC2": "428790",
      "lowCrossStreetB5SC3": "457390",
      "lowCrossStreetCode1": "40744001",
      "lowCrossStreetCode2": "42879001",
      "lowCrossStreetCode3": "45739001",
      "lowCrossStreetName1": "27 AVENUE",
      "lowCrossStreetName2": "ASTORIA BOULEVARD",
      "lowCrossStreetName3": "NEWTOWN AVENUE",
      "lowHouseNumberOfBlockfaceSortFormat": "100027001AA",
      "lowHouseNumberOfDefiningAddressRange": "100027001AA",
      "message2": "ADDR NUMBER ALTERED: HYPHEN INSERTED",
      "nta": "QN71",
      "ntaName": "Old Astoria",
      "numberOfCrossStreetB5SCsHighAddressEnd": "1",
      "numberOfCrossStreetB5SCsLowAddressEnd": "3",
      "numberOfCrossStreetsHighAddressEnd": "1",
      "numberOfCrossStreetsLowAddressEnd": "3",
      "numberOfEntriesInListOfGeographicIdentifiers": "0002",
      "numberOfExistingStructuresOnLot": "0001",
      "numberOfStreetFrontagesOfLot": "02",
      "physicalId": "0011640",
      "policePatrolBoroughCommand": "6",
      "policePrecinct": "114",
      "reasonCode1a": "2",
      "reasonCode2": "2",
      "returnCode1a": "01",
      "returnCode1e": "00",
      "roadwayType": "1",
      "rpadBuildingClassificationCode": "K2",
      "rpadSelfCheckCodeForBbl": "7",
      "sanbornBoroughCode": "4",
      "sanbornPageNumber": "015",
      "sanbornVolumeNumber": "02",
      "sanitationCollectionSchedulingSectionAndSubsection": "3C",
      "sanitationDistrict": "401",
      "sanitationRecyclingCollectionSchedule": "EM",
      "sanitationRegularCollectionSchedule": "MTH",
      "sanitationSnowPriorityCode": "C",
      "segmentAzimuth": "238",
      "segmentIdentifier": "0068622",
      "segmentLengthInFeet": "00267",
      "segmentOrientation": "3",
      "segmentTypeCode": "U",
      "sideOfStreetIndicator": "L",
      "sideOfStreetOfVanityAddress": "L",
      "splitLowHouseNumber": "100027001AA",
      "stateSenatorialDistrict": "12",
      "streetName1In": "21ST ST",
      "streetStatus": "2",
      "streetWidth": "60",
      "taxMapNumberSectionAndVolume": "40402",
      "toLionNodeId": "0042175",
      "toPreferredLgcsFirstSetOf5": "01",
      "trafficDirection": "T",
      "underlyingStreetCode": "40674001",
      "uspsPreferredCityName": "ASTORIA",
      "workAreaFormatIndicatorIn": "C",
      "xCoordinate": "1004755",
      "xCoordinateHighAddressEnd": "1004634",
      "xCoordinateLowAddressEnd": "1004774",
      "xCoordinateOfCenterofCurvature": "0000000",
      "yCoordinate": "0220605",
      "yCoordinateHighAddressEnd": "0220415",
      "yCoordinateLowAddressEnd": "0220641",
      "yCoordinateOfCenterofCurvature": "0000000",
      "zipCode": "11102",
    },
  },
  "google_response": Object {
    "plus_code": Object {
      "compound_code": "Q3CF+RJ New York, NY, USA",
      "global_code": "87G8Q3CF+RJ",
    },
    "results": Array [
      Object {
        "address_components": Array [
          Object {
            "long_name": "2701",
            "short_name": "2701",
            "types": Array [
              "street_number",
            ],
          },
          Object {
            "long_name": "21st Street",
            "short_name": "21st St",
            "types": Array [
              "route",
            ],
          },
          Object {
            "long_name": "Astoria",
            "short_name": "Astoria",
            "types": Array [
              "neighborhood",
              "political",
            ],
          },
          Object {
            "long_name": "Queens",
            "short_name": "Queens",
            "types": Array [
              "political",
              "sublocality",
              "sublocality_level_1",
            ],
          },
          Object {
            "long_name": "Queens County",
            "short_name": "Queens County",
            "types": Array [
              "administrative_area_level_2",
              "political",
            ],
          },
          Object {
            "long_name": "New York",
            "short_name": "NY",
            "types": Array [
              "administrative_area_level_1",
              "political",
            ],
          },
          Object {
            "long_name": "United States",
            "short_name": "US",
            "types": Array [
              "country",
              "political",
            ],
          },
          Object {
            "long_name": "11102",
            "short_name": "11102",
            "types": Array [
              "postal_code",
            ],
          },
        ],
        "formatted_address": "2701 21st St, Astoria, NY 11102, USA",
        "geometry": Object {
          "location": Object {
            "lat": 40.7719672,
            "lng": -73.92589,
          },
          "location_type": "ROOFTOP",
          "viewport": Object {
            "northeast": Object {
              "lat": 40.7733161802915,
              "lng": -73.9245410197085,
            },
            "southwest": Object {
              "lat": 40.7706182197085,
              "lng": -73.9272389802915,
            },
          },
        },
        "place_id": "ChIJpbB_M09fwokRsw6tOsOT3Wk",
        "plus_code": Object {
          "compound_code": "Q3CF+QJ New York, United States",
          "global_code": "87G8Q3CF+QJ",
        },
        "types": Array [
          "street_address",
        ],
      },
      Object {
        "address_components": Array [
          Object {
            "long_name": "27-8",
            "short_name": "27-8",
            "types": Array [
              "street_number",
            ],
          },
          Object {
            "long_name": "21st Street",
            "short_name": "21st St",
            "types": Array [
              "route",
            ],
          },
          Object {
            "long_name": "Astoria",
            "short_name": "Astoria",
            "types": Array [
              "neighborhood",
              "political",
            ],
          },
          Object {
            "long_name": "Queens",
            "short_name": "Queens",
            "types": Array [
              "political",
              "sublocality",
              "sublocality_level_1",
            ],
          },
          Object {
            "long_name": "Queens County",
            "short_name": "Queens County",
            "types": Array [
              "administrative_area_level_2",
              "political",
            ],
          },
          Object {
            "long_name": "New York",
            "short_name": "NY",
            "types": Array [
              "administrative_area_level_1",
              "political",
            ],
          },
          Object {
            "long_name": "United States",
            "short_name": "US",
            "types": Array [
              "country",
              "political",
            ],
          },
          Object {
            "long_name": "11102",
            "short_name": "11102",
            "types": Array [
              "postal_code",
            ],
          },
          Object {
            "long_name": "3634",
            "short_name": "3634",
            "types": Array [
              "postal_code_suffix",
            ],
          },
        ],
        "formatted_address": "27-8 21st St, Astoria, NY 11102, USA",
        "geometry": Object {
          "location": Object {
            "lat": 40.7720424,
            "lng": -73.92609569999999,
          },
          "location_type": "RANGE_INTERPOLATED",
          "viewport": Object {
            "northeast": Object {
              "lat": 40.77339138029149,
              "lng": -73.9247467197085,
            },
            "southwest": Object {
              "lat": 40.77069341970849,
              "lng": -73.9274446802915,
            },
          },
        },
        "place_id": "EiQyNy04IDIxc3QgU3QsIEFzdG9yaWEsIE5ZIDExMTAyLCBVU0EiGhIYChQKEgmZpcAzT1_CiREgUuBpLhXc_BAI",
        "types": Array [
          "street_address",
        ],
      },
    ],
    "status": "OK",
  },
  "lat": 40.772004329580426,
  "long": -73.92590668144963,
  "valid": true,
}
`;

exports[`validateLocation returns the right object 1`] = `
Object {
  "cb_data": Object {
    "address": "49-51 Chambers Street, Room 715
New York, NY 10007",
    "borough": "Manhattan",
    "email": "man01@cb.nyc.gov",
    "fax": "212-442-5055",
    "id": 101,
    "neighborhoods": "Tribeca, Seaport/Civic Center, Financial District, Battery Park City",
    "number": 1,
    "phone": "212-442-5050",
    "twitter": "CommunityBoard1",
    "website": "http://www.nyc.gov/html/mancb1/html/home/home.shtml",
  },
  "geoclient_response": Object {
    "address": Object {
      "assemblyDistrict": "66",
      "bbl": "1001237501",
      "bblBoroughCode": "1",
      "bblTaxBlock": "00123",
      "bblTaxLot": "7501",
      "boardOfElectionsPreferredLgc": "1",
      "boePreferredStreetName": "BROADWAY",
      "boePreferredstreetCode": "11361001",
      "boroughCode1In": "1",
      "buildingIdentificationNumber": "1087167",
      "businessImprovementDistrict": "113140",
      "censusBlock2000": "1008",
      "censusBlock2010": "1008",
      "censusTract1990": "  21  ",
      "censusTract2000": "  21  ",
      "censusTract2010": "  21  ",
      "cityCouncilDistrict": "01",
      "civilCourtDistrict": "01",
      "coincidentSegmentCount": "1",
      "communityDistrict": "101",
      "communityDistrictBoroughCode": "1",
      "communityDistrictNumber": "01",
      "communitySchoolDistrict": "02",
      "condominiumBillingBbl": "1001237501",
      "condominiumFlag": "C",
      "congressionalDistrict": "10",
      "continuousParityIndicator1a": "L",
      "continuousParityIndicator1e": "L",
      "cooperativeIdNumber": "0000",
      "crossStreetNamesFlagIn": "E",
      "dcpPreferredLgc": "01",
      "dofCondominiumIdentificationNumber": "1514",
      "dotStreetLightContractorArea": "1",
      "dynamicBlock": "109",
      "electionDistrict": "004",
      "fireBattalion": "01",
      "fireCompanyNumber": "010",
      "fireCompanyType": "L",
      "fireDivision": "01",
      "firstBoroughName": "MANHATTAN",
      "firstStreetCode": "11361001010",
      "firstStreetNameNormalized": "BROADWAY",
      "fromLionNodeId": "0015354",
      "fromPreferredLgcsFirstSetOf5": "0101",
      "genericId": "0000386",
      "geosupportFunctionCode": "1B",
      "geosupportReturnCode": "00",
      "geosupportReturnCode2": "00",
      "gi5DigitStreetCode1": "13610",
      "gi5DigitStreetCode2": "27880",
      "gi5DigitStreetCode3": "12520",
      "giBoroughCode1": "1",
      "giBoroughCode2": "1",
      "giBoroughCode3": "1",
      "giBuildingIdentificationNumber1": "1087167",
      "giBuildingIdentificationNumber2": "1087167",
      "giBuildingIdentificationNumber3": "1087167",
      "giDcpPreferredLgc1": "01",
      "giDcpPreferredLgc2": "01",
      "giDcpPreferredLgc3": "01",
      "giHighHouseNumber1": "237",
      "giHighHouseNumber2": "20",
      "giHighHouseNumber3": "19",
      "giLowHouseNumber1": "229",
      "giLowHouseNumber2": "2",
      "giLowHouseNumber3": "1",
      "giSideOfStreetIndicator1": "L",
      "giSideOfStreetIndicator2": "L",
      "giSideOfStreetIndicator3": "R",
      "giStreetCode1": "11361001",
      "giStreetCode2": "12788001",
      "giStreetCode3": "11252001",
      "giStreetName1": "BROADWAY",
      "giStreetName2": "PARK PLACE",
      "giStreetName3": "BARCLAY STREET",
      "healthArea": "7700",
      "healthCenterDistrict": "15",
      "highBblOfThisBuildingsCondominiumUnits": "1001231013",
      "highCrossStreetB5SC1": "127880",
      "highCrossStreetCode1": "12788001",
      "highCrossStreetName1": "PARK PLACE",
      "highHouseNumberOfBlockfaceSortFormat": "000238000AA",
      "houseNumber": "230",
      "houseNumberIn": "230",
      "houseNumberSortFormat": "000230000AA",
      "hurricaneEvacuationZone": "5",
      "instructionalRegion": "MS",
      "interimAssistanceEligibilityIndicator": "I",
      "internalLabelXCoordinate": "0981941",
      "internalLabelYCoordinate": "0198844",
      "latitude": 40.7121526074704,
      "latitudeInternalLabel": 40.7124572517608,
      "legacySegmentId": "0023364",
      "lionBoroughCode": "1",
      "lionBoroughCodeForVanityAddress": "1",
      "lionFaceCode": "0755",
      "lionFaceCodeForVanityAddress": "0755",
      "lionKey": "1075501170",
      "lionKeyForVanityAddress": "1075501170",
      "lionSequenceNumber": "01170",
      "lionSequenceNumberForVanityAddress": "01170",
      "listOf4Lgcs": "0107",
      "longitude": -74.00798614053389,
      "longitudeInternalLabel": -74.0083288540082,
      "lowBblOfThisBuildingsCondominiumUnits": "1001231001",
      "lowCrossStreetB5SC1": "112520",
      "lowCrossStreetB5SC2": "127890",
      "lowCrossStreetCode1": "11252001",
      "lowCrossStreetCode2": "12789001",
      "lowCrossStreetName1": "BARCLAY STREET",
      "lowCrossStreetName2": "PARK ROW",
      "lowHouseNumberOfBlockfaceSortFormat": "000228000AA",
      "lowHouseNumberOfDefiningAddressRange": "000230000AA",
      "nta": "MN24",
      "ntaName": "SoHo-TriBeCa-Civic Center-Little Italy",
      "numberOfCrossStreetB5SCsHighAddressEnd": "1",
      "numberOfCrossStreetB5SCsLowAddressEnd": "2",
      "numberOfCrossStreetsHighAddressEnd": "1",
      "numberOfCrossStreetsLowAddressEnd": "2",
      "numberOfEntriesInListOfGeographicIdentifiers": "0003",
      "numberOfExistingStructuresOnLot": "0001",
      "numberOfStreetFrontagesOfLot": "03",
      "physicalId": "0000499",
      "policePatrolBoroughCommand": "1",
      "policePrecinct": "001",
      "returnCode1a": "00",
      "returnCode1e": "00",
      "roadwayType": "1",
      "rpadBuildingClassificationCode": "RB",
      "rpadSelfCheckCodeForBbl": "0",
      "sanbornBoroughCode": "1",
      "sanbornPageNumber": "019",
      "sanbornVolumeNumber": "1S",
      "sanitationCollectionSchedulingSectionAndSubsection": "1A",
      "sanitationDistrict": "101",
      "sanitationRecyclingCollectionSchedule": "EF",
      "sanitationRegularCollectionSchedule": "MWF",
      "sanitationSnowPriorityCode": "C",
      "segmentAzimuth": "058",
      "segmentIdentifier": "0023364",
      "segmentLengthInFeet": "00230",
      "segmentOrientation": "N",
      "segmentTypeCode": "U",
      "selfCheckCodeOfBillingBbl": "0",
      "sideOfStreetIndicator": "L",
      "sideOfStreetOfVanityAddress": "L",
      "splitLowHouseNumber": "000228000AA",
      "stateSenatorialDistrict": "26",
      "streetName1In": "BROADWAY",
      "streetStatus": "2",
      "streetWidth": "52",
      "taxMapNumberSectionAndVolume": "10104",
      "toLionNodeId": "0015357",
      "toPreferredLgcsFirstSetOf5": "01",
      "trafficDirection": "A",
      "underlyingStreetCode": "11361001",
      "uspsPreferredCityName": "NEW YORK",
      "workAreaFormatIndicatorIn": "C",
      "xCoordinate": "0982036",
      "xCoordinateHighAddressEnd": "0982125",
      "xCoordinateLowAddressEnd": "0982004",
      "xCoordinateOfCenterofCurvature": "0000000",
      "yCoordinate": "0198733",
      "yCoordinateHighAddressEnd": "0198871",
      "yCoordinateLowAddressEnd": "0198675",
      "yCoordinateOfCenterofCurvature": "0000000",
      "zipCode": "10279",
    },
  },
  "google_response": Object {
    "plus_code": Object {
      "compound_code": "PX7V+4H New York, NY, USA",
      "global_code": "87G7PX7V+4H",
    },
    "results": Array [
      Object {
        "address_components": Array [
          Object {
            "long_name": "230",
            "short_name": "230",
            "types": Array [
              "street_number",
            ],
          },
          Object {
            "long_name": "Broadway",
            "short_name": "Broadway",
            "types": Array [
              "route",
            ],
          },
          Object {
            "long_name": "Manhattan",
            "short_name": "Manhattan",
            "types": Array [
              "political",
              "sublocality",
              "sublocality_level_1",
            ],
          },
          Object {
            "long_name": "New York",
            "short_name": "New York",
            "types": Array [
              "locality",
              "political",
            ],
          },
          Object {
            "long_name": "New York County",
            "short_name": "New York County",
            "types": Array [
              "administrative_area_level_2",
              "political",
            ],
          },
          Object {
            "long_name": "New York",
            "short_name": "NY",
            "types": Array [
              "administrative_area_level_1",
              "political",
            ],
          },
          Object {
            "long_name": "United States",
            "short_name": "US",
            "types": Array [
              "country",
              "political",
            ],
          },
          Object {
            "long_name": "10007",
            "short_name": "10007",
            "types": Array [
              "postal_code",
            ],
          },
        ],
        "formatted_address": "230 Broadway, New York, NY 10007, USA",
        "geometry": Object {
          "location": Object {
            "lat": 40.7127421,
            "lng": -74.00596890000001,
          },
          "location_type": "ROOFTOP",
          "viewport": Object {
            "northeast": Object {
              "lat": 40.7140910802915,
              "lng": -74.00461991970852,
            },
            "southwest": Object {
              "lat": 40.7113931197085,
              "lng": -74.00731788029152,
            },
          },
        },
        "place_id": "ChIJVzXOjBhawokRSp5VqCM4F9Q",
        "plus_code": Object {
          "compound_code": "PX7V+3J New York, United States",
          "global_code": "87G7PX7V+3J",
        },
        "types": Array [
          "street_address",
        ],
      },
    ],
    "status": "OK",
  },
  "lat": 40.7128,
  "long": -74.006,
  "valid": true,
}
`;
